### ISSUE TITLE
Downloading Symbols that are already in the .alPackages folder.

### ISSUE THREAD

**Issue by Spycoclown** [microsoft/navcontainerhelper#1299]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
i get an issue that it cant not download the app file. (Symbols)
It don't need to download it because it is already there.

**Scripts used to create container and cause the issue**
```
 Compile-AppInBCContainer -containerName $containerName -credential $adminCredentials -appProjectFolder $appProjectFolder -appSymbolsFolder $buildSymbolsFolder -appOutputFolder (Join-Path $buildArtifactFolder $_) -UpdateSymbols:$false -AzureDevOps:$true -CopyAppToSymbolsFolder
```

**Full output of scripts**
```
C:\agent-01\_work\91\a\app\App1-System_1.0.0.1399.app successfully created in 76 seconds
App1-System_1.0.0.1399.app copied to C:\agent-01\_work\91\s\App1-System\.alPackages
C:\agent-01\_work\91\a\app\App1-System_1.0.0.1399.app
Using Version 1.0.0.1399
Compiling folder app test
Using Symbols Folder: C:\agent-01\_work\91\s\App1-System\.alPackages
Disabling SSL Verification
Downloading symbols: App1-System_1.0.0.0.app <-- why???
Url : https://123.22.153.189:7049/BC/dev/packages?publisher=GWS%20mbH&appName=App1-System&versionText=1.0.0.0&tenant=default
##[error]Compile-AppInBCContainer : Exception calling "DownloadFile" with "2" argument(s): "The remote server returned an 
error: (404) Not Found."
At C:\agent-01\_work\91\s\Pipeline\Build-App\Compile-App.ps1:53 char:5
+     Compile-AppInBCContainer -containerName $containerName -credentia ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Compile-AppInBcContainer], MethodInvocationException
    + FullyQualifiedErrorId : WebException,Compile-AppInBcContainer

```

**Comment by freddydk** [comment]
-updatesymbols will always try to download symbols - try to remove that flag.

**Comment by Spycoclown** [comment]
Sry my mistake it is $false not true.
```
Compile-AppInBCContainer -containerName $containerName -credential $adminCredentials -appProjectFolder $appProjectFolder -appSymbolsFolder $buildSymbolsFolder -appOutputFolder (Join-Path $buildArtifactFolder $_) -UpdateSymbols:$false -AzureDevOps:$true -CopyAppToSymbolsFolder
```

**Comment by Spycoclown** [comment]
BCContainerhelper:
´´´
Version    Name                                Repository           Description                                                                                                                                                                          
-------    ----                                ----------           -----------                                                                                                                                                                          
1.0.6      BcContainerHelper                   PSGallery            PowerShell module                                                                                                                                                                    
´´´

**Comment by freddydk** [comment]
Strange - I am doing the same thing myself - copying the app I just compiled to the folder and using that.
There must be some difference between the app you compile and the one you have a dependency to.
I would need something I can repro from if I should be able to help.

**Comment by Spycoclown** [comment]
I could alocade the problem:
The shared folder are not synced.

I tryed this code:
```
write-host "1: Outside Container $appSymbolsFolder"
Get-ChildItem -Path $appSymbolsFolder

$existingApps = Invoke-ScriptInBcContainer -containerName $containerName -ScriptBlock { Param($appSymbolsFolder)
    write-host "2: Inside Container"
    Get-ChildItem -Path (Join-Path $appSymbolsFolder '*.app') | ForEach-Object { 
        write-host "Inside ""$($_.FullName)""" 
        Get-NavAppInfo -Path $_.FullName
    }
} -ArgumentList $containerSymbolsFolder

write-host "3:  Outside Container $containerSymbolsFolder"
Get-ChildItem -Path $appSymbolsFolder | ForEach-Object {
        write-host "Outside ""$($_.FullName)""" 
}

Invoke-ScriptInBcContainer -containerName $containerName -ScriptBlock { 
    Param($appSymbolsFolder, $appName)
    write-host "4: Inside Container"
    Get-ChildItem -Path $appSymbolsFolder | ForEach-Object {
            write-host "Inside $($_.FullName)"  
    }
    if (Test-Path -Path (Join-Path -Path $appSymbolsFolder -ChildPath $appName)) {
        Write-Host "${appName} copied to ${appSymbolsFolder}"
    }
} -ArgumentList $containerSymbolsFolder,"$($appName)"
```
the output was:
![image](https://user-images.githubusercontent.com/7090519/93986891-2ddb9900-fd87-11ea-83f1-0573487bdc30.png)

if i run the code after the pipeline is finish, it works all great the app is synced to the container.

**Comment by freddydk** [comment]
How is the container created?
Could you post script and full output of container creation (which shares do you have)

**Comment by Spycoclown** [comment]
Code:
```
Param(
    [Parameter(Mandatory = $true)]
    [string] $containerName,
    
    [Parameter(Mandatory = $true)]
    [string] $sharedFolder,
    
    [Parameter(Mandatory = $true)]
    [string] $bcImage,
    
    [Parameter(Mandatory = $true)]
    [string] $artifactUrl,
    
    [Parameter(Mandatory = $true)]
    [PSCredential] $adminCredentials,

    [Parameter(Mandatory = $true)]
    [string] $workingDirectory,

    [Parameter(Mandatory = $true)]
    $cleanContainer,

    [Parameter(Mandatory = $false)]
    [string] $license = 'fin.flf'
)

Write-Host "Creating needed parameters."
$shortcuts = "None"

$licensefile = Join-Path $workingDirectory $license

$additionalParameters = @("--volume ""$($sharedFolder):$($sharedFolder)""",
                          "--env customNavSettings=EnableTaskScheduler=$false")
    
$cleanContainer = [System.Convert]::ToBoolean($cleanContainer)
if($cleanContainer){
    $bcImage = ''
    $includeTestToolkit = $true
} else {
    $artifactUrl = ''
    $includeTestToolkit = $false
}

Write-Host "Starting to create container $containerName"
New-BCContainer -accept_eula `
   -containerName $containerName `
    -auth NAVUserPassword `
    -Credential $adminCredentials `
   -imageName $bcImage `
    -artifactUrl $artifactUrl `
    -additionalParameters $additionalParameters `
    -licenseFile $licensefile `
    -shortcuts $shortcuts `
    -includeTestToolkit:$includeTestToolkit `
    -doNotUseRuntimePackages `
    -assignPremiumPlan `
    -accept_outdated `
    -memoryLimit 8G `
    -multitenant:$false `
    -updateHosts `
    -useTraefik `
    -PublicDnsName $env:DNSNAME `
    -Restart always
```


Output of Container Creating:
```
2020-09-23T07:55:32.0596437Z ##[section]Starting: Create Container
2020-09-23T07:55:32.0873879Z ==============================================================================
2020-09-23T07:55:32.0874303Z Task         : PowerShell
2020-09-23T07:55:32.0874647Z Description  : Run a PowerShell script on Linux, macOS, or Windows
2020-09-23T07:55:32.0874959Z Version      : 2.170.1
2020-09-23T07:55:32.0875247Z Author       : Microsoft Corporation
2020-09-23T07:55:32.0875650Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
2020-09-23T07:55:32.0876109Z ==============================================================================
2020-09-23T07:55:33.4681068Z Generating script.
2020-09-23T07:55:33.4753611Z Formatted command: . 'C:\agent-01\_work\91\s\Pipeline\Create-Environment\Create-Container.ps1' -containerName Agent-01 -sharedFolder C:\agent-01 -bcImage sandbox1641469316035de:TestToolkit -artifactUrl "https://bcartifacts.azureedge.net/sandbox/16.4.14693.16035/de" -adminCredentials ([PSCredential]::new("app2Admin", (ConvertTo-SecureString -String "app12Admin" -AsPlainText -Force))) -workingDirectory C:\agent-01\_work\91\s -license fin.flf -cleanContainer False
2020-09-23T07:55:33.5384213Z ========================== Starting Command Output ===========================
2020-09-23T07:55:33.5833777Z ##[command]"C:\windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'C:\agent-01\_work\_temp\5e31d9d0-c23e-47d9-948d-706b22583849.ps1'"
2020-09-23T07:55:34.0598090Z Creating needed parameters.
2020-09-23T07:55:34.0968921Z Starting to create container Agent-01
2020-09-23T07:55:37.6683713Z BcContainerHelper is version 1.0.6
2020-09-23T07:55:37.6697615Z BcContainerHelper is running as administrator
2020-09-23T07:55:37.6845917Z Host is Microsoft Windows Server 2019 Datacenter - ltsc2019
2020-09-23T07:55:37.8600637Z Docker Client Version is 19.03.12
2020-09-23T07:55:37.8615767Z Docker Server Version is 19.03.12
2020-09-23T07:55:38.9478590Z Removing container Agent-01
2020-09-23T07:55:40.2249074Z Removing agent from host hosts file
2020-09-23T07:55:40.3070465Z Removing agent-* from host hosts file
2020-09-23T07:55:40.3732356Z Removing C:\ProgramData\BcContainerHelper\Extensions\Agent-01
2020-09-23T07:55:40.4092626Z Fetching all docker images
2020-09-23T07:55:40.5889091Z Enabling SSL as otherwise all clients will see mixed HTTP / HTTPS request, which will cause problems e.g. on the mobile and modern windows clients
2020-09-23T07:55:40.6093156Z Using image sandbox1641469316035de:testtoolkit
2020-09-23T07:55:40.7646143Z PublicDnsName is app.westeurope.cloudapp.azure.com
2020-09-23T07:55:40.7774474Z Creating Container Agent-01
2020-09-23T07:55:40.7880649Z Version: 16.4.14693.16035-DE
2020-09-23T07:55:40.7893890Z Style: sandbox
2020-09-23T07:55:40.7906085Z Multitenant: No
2020-09-23T07:55:40.8015712Z Platform: 16.0.14648.15997
2020-09-23T07:55:40.8043213Z Generic Tag: 0.1.0.21
2020-09-23T07:55:40.8078805Z Container OS Version: 10.0.17763.1457 (ltsc2019)
2020-09-23T07:55:40.8091797Z Host OS Version: 10.0.17763.1490 (ltsc2019)
2020-09-23T07:55:41.8300550Z Using hyperv isolation
2020-09-23T07:55:41.8345713Z Using locale de-DE
2020-09-23T07:55:41.8377562Z Adding special CheckHealth.ps1 to enable Traefik support
2020-09-23T07:55:41.8405239Z Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
2020-09-23T07:55:41.8736876Z Using license file C:\agent-01\_work\91\s\fin.flf
2020-09-23T07:55:41.9525120Z Additional Parameters:
2020-09-23T07:55:41.9568847Z --volume "C:\agent-01:C:\agent-01"
2020-09-23T07:55:41.9581170Z --env customNavSettings=EnableTaskScheduler=False,PublicODataBaseUrl=https://app.westeurope.cloudapp.azure.com/Agent-01rest/odata,PublicSOAPBaseUrl=https://app.westeurope.cloudapp.azure.com/Agent-01soap/ws,PublicWebBaseUrl=https://app.westeurope.cloudapp.azure.com/Agent-01
2020-09-23T07:55:41.9592942Z --hostname agent
2020-09-23T07:55:41.9608165Z -e webserverinstance=Agent-01
2020-09-23T07:55:41.9622744Z -e publicdnsname=app.westeurope.cloudapp.azure.com
2020-09-23T07:55:41.9636183Z -l "traefik.protocol=https"
2020-09-23T07:55:41.9649752Z -l "traefik.web.frontend.rule=PathPrefix:/Agent-01"
2020-09-23T07:55:41.9662526Z -l "traefik.web.port=443"
2020-09-23T07:55:41.9676076Z -l "traefik.soap.frontend.rule=PathPrefix:/Agent-01soap;ReplacePathRegex: ^/Agent-01soap(.*) /BC$1"
2020-09-23T07:55:41.9688404Z -l "traefik.soap.port=7047"
2020-09-23T07:55:41.9701702Z -l "traefik.rest.frontend.rule=PathPrefix:/Agent-01rest;ReplacePathRegex: ^/Agent-01rest(.*) /BC$1"
2020-09-23T07:55:41.9735682Z -l "traefik.rest.port=7048"
2020-09-23T07:55:41.9737552Z -l "traefik.dev.frontend.rule=PathPrefix:/Agent-01dev;ReplacePathRegex: ^/Agent-01dev(.*) /BC$1"
2020-09-23T07:55:41.9746041Z -l "traefik.dev.port=7049"
2020-09-23T07:55:41.9761498Z -l "traefik.dl.frontend.rule=PathPrefixStrip:/Agent-01dl"
2020-09-23T07:55:41.9774650Z -l "traefik.dl.port=8080"
2020-09-23T07:55:41.9787823Z -l "traefik.dl.protocol=http"
2020-09-23T07:55:41.9801564Z -l "traefik.enable=true"
2020-09-23T07:55:41.9814579Z -l "traefik.frontend.entryPoints=https"
2020-09-23T07:55:41.9827519Z Files in C:\ProgramData\BcContainerHelper\Extensions\Agent-01\my:
2020-09-23T07:55:42.0020742Z - AdditionalOutput.ps1
2020-09-23T07:55:42.0034525Z - CheckHealth.ps1
2020-09-23T07:55:42.0046824Z - license.flf
2020-09-23T07:55:42.0063641Z - MainLoop.ps1
2020-09-23T07:55:42.0076434Z - SetupNavUsers.ps1
2020-09-23T07:55:42.0089635Z - SetupVariables.ps1
2020-09-23T07:55:42.0101274Z - updatehosts.ps1
2020-09-23T07:55:42.0113638Z Creating container Agent-01 from image sandbox1641469316035de:testtoolkit
2020-09-23T07:55:42.2671901Z 14f5cb3c8df6e75ad4164dc0cd8f6ee25aeb0ae90476b882bfda674f530d8517
2020-09-23T07:55:49.3345434Z Waiting for container Agent-01 to be ready
2020-09-23T07:55:56.4261481Z Initializing...
2020-09-23T07:55:56.4273935Z Setting host.containerhelper.internal to 172.24.48.1 in container hosts file
2020-09-23T07:55:56.4284723Z Starting Container
2020-09-23T07:55:56.4297096Z Hostname is agent
2020-09-23T07:55:57.5878524Z PublicDnsName is app.westeurope.cloudapp.azure.com
2020-09-23T07:55:58.7480358Z Using NavUserPassword Authentication
2020-09-23T07:56:01.1031242Z Starting Local SQL Server
2020-09-23T07:56:02.2951210Z Starting Internet Information Server
2020-09-23T07:56:03.4686678Z Creating Self Signed Certificate
2020-09-23T07:56:04.6461921Z Self Signed Certificate Thumbprint 256A4881FAF841082877F183A08D51D45AE1F3DB
2020-09-23T07:56:04.6476275Z Modifying Service Tier Config File with Instance Specific Settings
2020-09-23T07:56:04.6488528Z Modifying Service Tier Config File with settings from environment variable
2020-09-23T07:56:04.6501250Z Setting EnableTaskScheduler to False
2020-09-23T07:56:04.6525280Z Setting PublicODataBaseUrl to https://app.westeurope.cloudapp.azure.com/Agent-01rest/odata
2020-09-23T07:56:04.6526629Z Setting PublicSOAPBaseUrl to https://app.westeurope.cloudapp.azure.com/Agent-01soap/ws
2020-09-23T07:56:04.6540748Z Setting PublicWebBaseUrl to https://app.westeurope.cloudapp.azure.com/Agent-01
2020-09-23T07:57:36.6918828Z Starting Service Tier
2020-09-23T07:57:36.6953680Z Registering event sources
2020-09-23T07:57:48.6741445Z Creating DotNetCore Web Server Instance
2020-09-23T07:57:48.6752765Z Enabling Financials User Experience
2020-09-23T07:57:48.6764476Z Using license file 'c:\run\my\license.flf'
2020-09-23T07:57:50.9842608Z Import License
2020-09-23T07:57:52.1577081Z Creating http download site
2020-09-23T07:57:53.4783615Z Setting SA Password and enabling SA
2020-09-23T07:57:54.6493264Z Creating app2Admin as SQL User and add to sysadmin
2020-09-23T07:57:54.6505014Z Assign Premium plan for GLO
2020-09-23T07:57:54.6516461Z Assign Premium plan for CSC
2020-09-23T07:57:54.6529701Z Assign Premium plan for SMB
2020-09-23T07:57:54.6540999Z Assign Premium plan for THK
2020-09-23T07:57:54.6552835Z Assign Premium plan for DFE
2020-09-23T07:57:54.6563923Z Assign Premium plan for SEU
2020-09-23T07:57:54.6574941Z Assign Premium plan for BOE
2020-09-23T07:57:54.6585892Z Assign Premium plan for app2ADMIN
2020-09-23T07:57:55.8297429Z Assign Premium plan for RIP
2020-09-23T07:57:55.8312823Z Container IP Address: 173.34.50.123
2020-09-23T07:57:55.8326117Z Container Hostname  : agent
2020-09-23T07:57:55.8339801Z Container Dns Name  : app.westeurope.cloudapp.azure.com
2020-09-23T07:57:55.8352602Z Web Client          : https://app.westeurope.cloudapp.azure.com/Agent-01/
2020-09-23T07:57:55.8364668Z Dev. Server         : https://app.westeurope.cloudapp.azure.com
2020-09-23T07:57:55.8376691Z Dev. ServerInstance : BC
2020-09-23T07:57:55.8388698Z Setting agent to 173.34.50.123 in host hosts file
2020-09-23T07:57:55.8395086Z 
2020-09-23T07:57:55.8407821Z Files:
2020-09-23T07:57:55.8419604Z http://app.westeurope.cloudapp.azure.com:8080/ALLanguage.vsix
2020-09-23T07:57:55.8432143Z http://app.westeurope.cloudapp.azure.com:8080/certificate.cer
2020-09-23T07:57:55.8436645Z 
2020-09-23T07:57:55.8448551Z Initialization took 121 seconds
2020-09-23T07:57:55.8460523Z Ready for connections!
2020-09-23T07:57:55.8473608Z Reading CustomSettings.config from Agent-01
2020-09-23T07:57:57.1189478Z Container Agent-01 successfully created
2020-09-23T07:57:57.1202539Z Because of Traefik, the following URLs need to be used when accessing the container from outside your Docker host:
2020-09-23T07:57:57.1217229Z Web Client:        https://app.westeurope.cloudapp.azure.com/Agent-01
2020-09-23T07:57:57.1229426Z SOAP WebServices:  https://app.westeurope.cloudapp.azure.com/Agent-01soap
2020-09-23T07:57:57.1241045Z OData WebServices: https://app.westeurope.cloudapp.azure.com/Agent-01rest
2020-09-23T07:57:57.1251880Z Dev Service:       https://app.westeurope.cloudapp.azure.com/Agent-01dev
2020-09-23T07:57:57.1263650Z File downloads:    https://app.westeurope.cloudapp.azure.com/Agent-01dl
2020-09-23T07:57:57.1268412Z 
2020-09-23T07:57:57.1281508Z Use:
2020-09-23T07:57:57.1300501Z Get-BcContainerEventLog -containerName Agent-01 to retrieve a snapshot of the event log from the container
2020-09-23T07:57:57.1327497Z Get-BcContainerDebugInfo -containerName Agent-01 to get debug information about the container
2020-09-23T07:57:57.1361495Z Enter-BcContainer -containerName Agent-01 to open a PowerShell prompt inside the container
2020-09-23T07:57:57.1386901Z Remove-BcContainer -containerName Agent-01 to remove the container again
2020-09-23T07:57:57.1406947Z docker logs Agent-01 to retrieve information about URL's again
2020-09-23T07:57:57.2840750Z ##[section]Finishing: Create Container
```

**Comment by freddydk** [comment]
I think the issue is that you share the entire C:\agent-01 - I think you should share C:\agent-01\_work\91\s to c:\sources in the container or like (that's what I normally do)

**Comment by Spycoclown** [comment]
ok i will have a try

**Comment by freddydk** [comment]
In general, I think it is a timing issue - like when we download symbols, we actually wait for them to be available inside the container here: https://github.com/microsoft/navcontainerhelper/blob/5e50f98289ecbd53d895aa7f4735fe504464c19b/AppHandling/Compile-AppInNavContainer.ps1#L305

**Comment by Spycoclown** [comment]
ok, i'm not sure why, but i delete the hole vm and rebuild everything, now with the same code i don't get any error anymore.

The problem started when i changed the azure VM Size from B8ms to D8sv3. Now i rebuild with D8sv3 and it work well.

Thanks any way, maybe docker dont like size changes.

**Comment by freddydk** [comment]
It is a timing issue - it takes a little time for a file to become available - especially when you share a very large folder.
You can also insert a short sleep and see that the file is there after copying it - to be sure.

**Comment by Spycoclown** [comment]
i can't reproduce this error, i'm sorry.

**Comment by Spycoclown** [comment]
Ok i just tryed it again, and you are right. it is because i share a to large folder.

**Comment by Spycoclown** [comment]
Sry, but it still recome... i hope it was gone.
i will try to raise the delay.

**Comment by freddydk** [comment]
Will be in 1.0.7-preview sometime tonight/tomorrow.
Thanks

**Comment by Spycoclown** [comment]
Oh sorry I wasn't fast enough, thanks for finishing the request.

**Comment by freddydk** [comment]
np - had the same problem with another partner at 11pm yesterday...

**Comment by freddydk** [comment]
shipped in 1.0.7-preview191
